### DISCUSSION TITLE
Help with action buttons

### DISCUSSION THREAD

**Discussion by gldtn** [Power-Components/livewire-powergrid#389] [Q&A]
How do I translate this..
`<button wire:click="edit({{ $student->id }})">Edit</button>`

To:
Button::add('edit')
               ->caption('Edit')
               ->class('text-indigo-500 cursor-pointer px-3 py-2 m-1 rounded text-sm')
               ->emitTo('contact.crud', 'edit', ['$contact' => 'id']), <-- I tried emit and emitTo

This is my Livewire component App\Http\Livewire\contact\crud.php

public function edit($id)
    {
        $contact = Contact::findOrFail($id);
        $this->contact_id = $id;
        $this->fullname = $contact->fullname;
        $this->email = $contact->email;

        $this->openModalPopover();
    }

Any help is appreciated, thanks!

**Comment by dens4t** [comment]
I use assign the listener in the component, so that will be `protected $listeners = ['edit'=>'edit'`];
So in action table, i use `->emit('edit', ['key' => 'id'])`
